src/lib/components/DeleteButton.js

```diff
@@ -113,19 +113,19 @@ export class DeleteButtonComponent extends Component {
             <Button onClick={this.closeConfirmModal} floated="left">
               {i18next.t('Cancel')}
             </Button>
-            <ActionButton {...color} name="delete" onClick={this.handleDelete}>
-              {(formik) => (
-                <>
-                  {formik.isSubmitting &&
-                  actionState === DRAFT_DELETE_STARTED ? (
-                    <Icon size="large" loading name="spinner" />
-                  ) : (
-                    <Icon name="trash alternate outline" />
-                  )}
-                  {capitalizedActionLbl}
-                </>
-              )}
-            </ActionButton>
+            <ActionButton
+              {...color}
+              name="delete"
+              onClick={this.handleDelete}
+              icon={(formik) =>
+                formik.isSubmitting && actionState === DRAFT_DELETE_STARTED ? (
+                  <Icon size="large" loading name="spinner" />
+                ) : (
+                  <Icon name="trash alternate outline" />
+                )
+              }
+              content={capitalizedActionLbl }
```

Comment:
can this function be extracted outside the return clause? you would avoid passing callables to the icon prop